nova/scheduler/weights/affinity.py

```diff
@@ -33,8 +33,15 @@
 
 
 class _SoftAffinityWeigherBase(weights.BaseHostWeigher):
+    _SHARD_PREFIX = 'vc-'
     policy_name = None
 
+    def _is_host_in_aggregates(self, host, aggs):
+        for agg in aggs:
+            if host in agg.hosts:
+                return True
+        return False
```

Comment:
Can we use set-operations and then not have to for-loop over all the aggregates all the time.